be/src/util/exception.c

```diff
@@ -0,0 +1,225 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Copied from:
+//   https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/exception.c
+// We need Hadoop 2.7 for Hortonworks HDP support, but libhdfs no longer exports
+// getJNIEnv so we have to copy some source files like this to Impala. ROCANA-3465
```

Comment:
Good question.
In the commit comment I mention which files did not change, meaning they were an exact duplicate of the HDFS files. The other files differ from HDFS, but only slightly. They had more functions than we needed so I did not pull all the functions in. I also adjusted the `#include`s.

We could put this as comments in the source files, but that documentation could get stale over time as we edit the files. On the other hand, maybe we'll never edit them.

So, any preference on how we document this?